meson.build

```diff
@@ -0,0 +1,63 @@
+project(
+  'netpanzer',
+  ['c', 'cpp'],
+  version: '0.9.0.06',
+  meson_version : '>= 0.56.2',
+  default_options: [
+    'warning_level=1',
+    'cpp_std=c++17'
+    ]
+  )
+
+# Setup compiler.
+cxx = meson.get_compiler('cpp')
+cc = meson.get_compiler('c')
+
+extra_flags = [
+  '-fno-common',
+#  '-fstack-protector-strong',
+#  '-Wformat-security',
+#  '-Wformat-overflow=2',
+#  '-Wshadow',
+#  '-Wstrict-overflow=5'
+# '-Wconversion'
+  ]
+
+add_project_arguments(cc.get_supported_arguments(extra_flags), language: 'c')
+add_project_arguments(cxx.get_supported_arguments(extra_flags), language: 'cpp')
+
+conf = configuration_data()
+conf.set_quoted('PACKAGE_VERSION', meson.project_version())
+config_h = configure_file(output : 'config.h', configuration : conf)
+
+deps = [
+  dependency('SDL2'),
+# SDL2 is case sensitive for the ttf library
+  dependency('SDL2_ttf'),
+  dependency('SDL2_mixer'),
+  dependency('physfs')
+]
```

Comment:
Aside: all 4 of these dependencies are available via:
```
meson wrap install ${dep}
```

which means that you can let this build without fuss on Windows where package managers don't exist, and replaces the previous "include a code copy of physfs inside the repository" approach.

Reply:
I'm trying this in https://github.com/netpanzer/netpanzer/pull/133 but the resulting binary always links dynamically to the dependencies. I'm not sure how to handle this so I've opened up a [discussion on the meson repo](https://github.com/mesonbuild/meson/discussions/12754)